docs/sources/setup/migrate/migrate-storage-clients/_index.md

```diff
@@ -0,0 +1,155 @@
+Refer to the [thanos storage reference](https://grafana.com/docs/loki/<LOKI_VERSION>/configure/#thanos_object_store_config) to view the complete list of supported storage providers and their configuration options.
+
+### Enabling the new storage clients
+1. Enable thanos storage clients by setting `use_thanos_objstore` to `true` in the `storage_config` section or by setting `-use-thanos-objstore` flag to true.
+1. When enabled, configuration under `storage_config.object_store` takes effect instead of legacy storage configs
+   ```yaml
+   # Uses the new storage clients for connecting to gcs backend
+   storage_config:
+      use_thanos_objstore: true # enable the new storage clients
+   object_store:
+      gcs: 
+         bucket: "example-bucket"
+   ```
+1. You can also configure the new clients in the common storage section if you prefer to use the common config section.
+   ```yaml
+   storage_config:
+      use_thanos_objstore: true # enable the new storage clients
+   common:
+      storage:
+         object_store:
+            gcs:
+            bucket: "example-bucket"
+   ```
+1. Ruler storage should be configured under `ruler_storage` section when using the new storage clients.
+   ```yaml
+   storage_config:
+      use_thanos_objstore: true # enable the new storage clients
+   ruler_storage:
+      backend: gcs
+      gcs:
+         bucket: "example-bucket"
+   ```
+
+{{< admonition type="note" >}}
+Following sections document the steps for migrating the clients. Only some of the commonly used parameters are listed here.
+For a complete list of supported parameters, refer to the [thanos storage reference](https://grafana.com/docs/loki/<LOKI_VERSION>/configure/#thanos_object_store_config) and compare that to what you are using in your setup.
+{{< /admonition >}}
+
+### GCS Storage Migration
+
+When migrating from the legacy GCS storage client to the new Thanos-based client, you'll need to update your configuration parameters as follows:
+
+{{< responsive-table >}}
+| Legacy Parameter | New Parameter | Required Changes |
+|-----------------|---------------|------------------|
+| `bucket_name` | `bucket_name` | No changes required |
+| `chunk_buffer_size` | `chunk_size_bytes` | Rename parameter to `chunk_size_bytes` |
+| `enable_retries` | `enable_retries` | No changes required |
+{{< /responsive-table >}}
+
+**Example configuration migration:**
+
+Legacy configuration:
+```yaml
+storage_config:
+  gcs:
+    bucket_name: example-bucket
+    chunk_buffer_size: 10MB
+    enable_retries: true
+```
+
+New configuration:
+```yaml
+storage_config:
+  use_thanos_objstore: true
+object_store:
+  gcs:
+    bucket_name: example-bucket
+    chunk_size_bytes: 10MB
+    enable_retries: true
+```
+
+### S3 Storage Migration
+
+When migrating from the legacy S3 storage client to the new Thanos-based client, you'll need to update your configuration parameters as follows:
+
+{{< responsive-table >}}
+| Legacy Parameter | New Parameter | Required Changes |
```

Comment:
Would it make sense to put these parameters in alphabetical order?

Reply:
i tried to list them in the order they appear in the configuration reference. we could do it in alphabetical order, but i am not sure if it improves things :)

Reply:
Ah, that makes sense.
I was trained to always alphabetize lists, unless there is a reason not to.